src/features/canvass/components/CanvassMapOverlays.tsx

```diff
@@ -96,14 +97,16 @@ const CanvassMapOverlays: FC<Props> = ({
         )}
         {selectedLocation && expanded && (
           <Suspense>
-            <LocationDialog
-              assignment={assignment}
-              location={selectedLocation}
-              onClose={() => {
-                setExpanded(false);
-              }}
-              orgId={assignment.organization_id}
-            />
+            <ZUIConfirmDialogProvider>
```

Comment:
Good catch that this requires a provider, but why put the provider so deep in the tree? It could go in the `ClientContext` with other providers, couldn't it?

Reply:
But isn't that just a z index problem, that can be solved by tweaking the z index values of dialogs and the nav stack/page?